lib/src/layer/quadractic_curve_layer.dart

```diff
@@ -0,0 +1,131 @@
+// ignore_for_file: public_member_api_docs, sort_constructors_first
+import 'package:flutter/widgets.dart';
+import 'package:flutter_map/plugin_api.dart';
+
+import 'package:latlong2/latlong.dart' hide Path;
+
+class QuadraticCurveMarker {
+  final LatLng startPoint;
+  final LatLng endPoint;
+  final LatLng handlePoint;
+
+  final Color color;
+  final double borderStrokeWidth;
+  final Color borderColor;
+
+  Offset start = Offset.zero;
+  Offset end = Offset.zero;
+  Offset handle = Offset.zero;
+
+  QuadraticCurveMarker({
+    required this.startPoint,
+    required this.endPoint,
+    required this.handlePoint,
+    this.color = const Color.fromARGB(255, 255, 0, 0),
+    this.borderStrokeWidth = 0.0,
+    this.borderColor = const Color(0xFFFFFF00),
+  });
+
+  @override
+  bool operator ==(covariant QuadraticCurveMarker other) {
+    if (identical(this, other)) return true;
+
+    return other.startPoint == startPoint &&
+        other.endPoint == endPoint &&
+        other.handlePoint == handlePoint &&
+        other.color == color &&
+        other.borderStrokeWidth == borderStrokeWidth &&
+        other.borderColor == borderColor &&
+        other.start == start &&
+        other.end == end &&
+        other.handle == handle;
+  }
+
+  @override
+  int get hashCode {
+    return startPoint.hashCode ^
+        endPoint.hashCode ^
+        handlePoint.hashCode ^
+        color.hashCode ^
+        borderStrokeWidth.hashCode ^
+        borderColor.hashCode ^
+        start.hashCode ^
+        end.hashCode ^
+        handle.hashCode;
+  }
+}
+
+class QuadraticCurveLayer extends StatelessWidget {
+  final List<QuadraticCurveMarker> curves;
+  const QuadraticCurveLayer({
+    super.key,
+    this.curves = const [],
+  });
+
+  @override
+  Widget build(BuildContext context) {
+    return LayoutBuilder(
+      builder: (BuildContext context, BoxConstraints bc) {
+        final size = Size(bc.maxWidth, bc.maxHeight);
+        final map = FlutterMapState.maybeOf(context)!;
+        final curveWidgets = <Widget>[];
+        for (final curve in curves) {
+          curve.start = map.getOffsetFromOrigin(curve.startPoint);
+          curve.end = map.getOffsetFromOrigin(curve.endPoint);
+          curve.handle = map.getOffsetFromOrigin(curve.handlePoint);
```

Comment:
These can also be computed in custom paint to allow for only final variables in QuadraticCurveMarker